### PR TITLE
All claims update all claims mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
1. Updates All Claims schema to latest version per what's in vets-website
2. Re-adds original 0781 schema that was removed from vets-website master (backend is already built for this schema), and conversely, removes the `ptsdIncidents` and `secondaryPtsdIncidents` arrays that were in vets-website master
3. Moves `form0781` and `form4142` definitions out of `definitions` and into `properties`, since these are both finished on the backend.
4. Reverts `definitions.ptsdIncident.date` and `definitions.secondaryPtsdIncident.date` to `incidentDate`, as this makes the presubmit translations easier for that part of the form since the api is already built.
5. Removes `minItems` property on `alternateNames` as that's a bug getting tracked on a separate ticket.

## Testing:
- Schema is valid
- All unit tests pass on `vets-website` when pointing at this schema locally
- Manual form testing

## AC:
- [ ] All Claims form works with no regressions when using this schema

### CODE REVIEW

**Comment by markgreenburg** [comment]
@randyhoffmastergit @robfelder @aaronjlech FYI, PR to update All Claims schema in vets-json-schema

File: src/schemas/21-526EZ-allclaims/schema.js [hunk 1]
```diff
@@ -34,6 +17,14 @@ const serviceBranches = [
   'Public Health Service'
 ];
 
+const specialIssues = {
```
Comment (cvalarida) [inline src/schemas/21-526EZ-allclaims/schema.js]: Can we define this in the schema's `definitions` only?

File: src/schemas/21-526EZ-allclaims/schema.js [hunk 2]
```diff
@@ -597,6 +666,72 @@ const schema = {
...
+                  },
+                  street2: {
+                    type: 'string',
+                    minLength: 1,
+                    maxLength: 20
+                  },
+                  city: {
+                    type: 'string',
+                    minLength: 1,
+                    maxLength: 30
```
Comment (cvalarida) [inline src/schemas/21-526EZ-allclaims/schema.js]: Still chuckling at this.

File: src/schemas/21-526EZ-allclaims/schema.js [hunk 3]
```diff
@@ -597,6 +666,72 @@ const schema = {
...
+                  },
+                  postalCode: {
+                    type: 'string',
+                    pattern: '^\\d{5}(?:([-\\s]?)\\d{4})?$'
+                  },
+                  country: {
+                    type: 'string',
+                    enum: baseAddressDef.properties.country.enum,
```
Comment (cvalarida) [inline src/schemas/21-526EZ-allclaims/schema.js]: Could we define `country` as `baseAddressDef.properties.country`?

File: src/schemas/21-526EZ-allclaims/schema.js (hunk 1 above)
Comment (markgreenburg) [reply src/schemas/21-526EZ-allclaims/schema.js]: ahh, nice catch

File: src/schemas/21-526EZ-allclaims/schema.js [hunk 4]
```diff
@@ -147,47 +126,111 @@ const schema = {
...
+      properties: {
+        authorities: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              name: {
+                type: 'string'
+              },
+              address: {
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/schemas/21-526EZ-allclaims/schema.js]: Yep, that's the one

File: src/schemas/21-526EZ-allclaims/schema.js [hunk 5]
```diff
@@ -147,47 +126,111 @@ const schema = {
...
+        description: { type: 'string' },
+        unitAssigned: { $ref: '#/definitions/unitAssigned' },
+        unitAssignedDates: { $ref: '#/definitions/unitAssignedDates' }
+      }
+    }
   },
   properties: {
     alternateNames: {
       type: 'array',
+      minItems: 1,
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/schemas/21-526EZ-allclaims/schema.js]: Probably not, I think this is the source of a bug I'm tracking on another issue.

File: src/schemas/21-526EZ-allclaims/schema.js (hunk 3 above)
Comment (markgreenburg) [reply src/schemas/21-526EZ-allclaims/schema.js]: Yeah, that'd make sense, wouldn't it? States as well.

File: src/schemas/21-526EZ-allclaims/schema.js (hunk 2 above)
Comment (markgreenburg) [reply src/schemas/21-526EZ-allclaims/schema.js]: envelopes 🤷‍♂️

File: src/schemas/21-526EZ-allclaims/schema.js [hunk 6]
```diff
@@ -481,42 +528,61 @@ const schema = {
       type: 'boolean',
       default: false
     },
-    ptsd781Attachment: {
```
[the remark this answers is not in the document]
Comment (markgreenburg) [reply src/schemas/21-526EZ-allclaims/schema.js]: This is gone from the schema in vets-website, so I'm assuming it's either getting actively developed or no longer needed.

**Pull request merged**